cluster/sync.go

```diff
@@ -3,34 +3,41 @@ package cluster
 import (
 	"strings"
 
+	"github.com/weaveworks/flux"
 	"github.com/weaveworks/flux/resource"
 )
 
 // Definitions for use in synchronising a cluster with a git repo.
 
-// SyncAction represents either the deletion or application (create or
-// update) of a resource.
-type SyncAction struct {
-	Delete resource.Resource // ) one of these
-	Apply  resource.Resource // )
+// SyncStack groups a set of resources to be updated. The purpose of
+// the grouping is to limit the "blast radius" of changes. For
+// example, if we calculate a checksum for each stack and annotate the
+// resources within it, any single change will affect only the
+// resources in the same stack, meaning fewer things to annotate. (So
+// why not do these individually? This, too, can be expensive, since
+// it involves examining each resource individually).
+type SyncStack struct {
```

Comment:
`SyncAction` was a really good name. On the other hand `Stack` (which I read about before getting here) was a bit confusing.  It made me wonder if it had to do something with user's application stack (or whether we are explicitly letting the user split their cluster resources in `Stacks`).

I probably don't have a better alternative. Maybe `SyncSet`? or ... `SyncGroup`? Or ... `SyncTargetGroup`?

Reply:
Also, I am wondering whether we can use Merkle Trees (like Cassandra uses for repairs) in order to make this more efficient.

Reply:
`SyncSet` is good. `Stack` comes from the inspiration for the PR, via the author -- I left it in, as homage, but the PR has drifted far enough that it's a bit misleading now.